src/sgm_stereo_node.cpp

```diff
@@ -315,20 +402,14 @@ void StereoSGMNode::callback(const sensor_msgs::ImageConstPtr& left_ros_img,
   toc("computeSGMStereoDisparity");
   pub_disparity_.publish(disp_msg);
 
-  sensor_msgs::PointCloud2 pcl_msg;
-  tic();
-  computePointCloudFromDisparity(left_ros_img, model_, disp_msg, pcl_msg);
-  toc("computePointCloudFromDisparity");
-  pub_pcl_.publish(pcl_msg);
+  if (publish_sgm_pcl_)
+  {
+    sensor_msgs::PointCloud2 pcl_msg;
+    tic();
+    computePointCloudFromDisparity(left_ros_img, model_, disp_msg, pcl_msg);
```

Comment:
The `stereo_img_proc` `points2` nodelet should be able to do this, so we should be able to remove it from here.

Did you try that?

Reply:
Yes see the launch file. I'll remove it from here.